custom/enikshay/integrations/nikshay/tests/test_repeaters.py

```diff
@@ -295,3 +301,151 @@ def test_handle_failure(self):
         self._assert_case_property_equal(updated_episode_case, 'nikshay_error', unicode(message))
+
+
+class TestNikshayHIVTestRepeater(ENikshayLocationStructureMixin, NikshayRepeaterTestBase):
+
+    def setUp(self):
+        super(TestNikshayHIVTestRepeater, self).setUp()
+
+        self.repeater = NikshayHIVTestRepeater(
+            domain=self.domain,
+            url='case-repeater-url',
+            username='test-user'
+        )
+        self.repeater.white_listed_case_types = ['person']
+        self.repeater.save()
+
+    def test_not_available_for_domain(self):
+        self.assertFalse(NikshayHIVTestRepeater.available_for_domain(self.domain))
+
+    @flag_enabled('NIKSHAY_INTEGRATION')
+    def test_available_for_domain(self):
+        self.assertTrue(NikshayHIVTestRepeater.available_for_domain(self.domain))
+
+    @run_with_all_backends
+    def test_trigger(self):
+        # nikshay not enabled
+        self.assertEqual(0, len(self.repeat_records().all()))
+
+        self.factory.create_or_update_cases([self.episode])
+        update_case(
+            self.domain,
+            self.episode_id,
+            {
+                "nikshay_registered": 'true',
+                "nikshay_id": DUMMY_NIKSHAY_ID,
+            },
+        )
+        update_case(
+            self.domain,
+            self.person_id,
+            {
+                "hiv_status": "unknown",
+                "owner_id": self.phi.location_id,
+            }
+        )
+        self.assertEqual(1, len(self.repeat_records().all()))
+        update_case(
+            self.domain,
+            self.person_id,
+            {
+                "hiv_status": "reactive",
+                "cpt_initiation_date": "2016-01-01"
+            }
+        )
+        self.assertEqual(2, len(self.repeat_records().all()))
+        update_case(
+            self.domain,
+            self.person_id,
+            {
+                "art_initiation_date": "2016-02-01"
+            }
+        )
+        self.assertEqual(3, len(self.repeat_records().all()))
+
+
+class TestNikshayHIVTestPayloadGenerator(ENikshayLocationStructureMixin, NikshayRepeaterTestBase):
+    def setUp(self):
+        super(TestNikshayHIVTestPayloadGenerator, self).setUp()
+        self.cases = self.create_case_structure()
+        self.person_case = self.cases['person']
+        self.episode_case = self.cases['episode']
+        self._create_nikshay_registered_case()
+
+        MockRepeater = namedtuple('MockRepeater', 'username password')
+        MockRepeatRecord = namedtuple('MockRepeatRecord', 'repeater')
+        self.repeat_record = MockRepeatRecord(MockRepeater(username="arwen", password="Hadhafang"))
+
+    def create_case_structure(self):
+        return {case.get_id: case for case in filter(None, self.factory.create_or_update_cases(
+            [self.person, self.episode]))}
+
+    def _create_nikshay_registered_case(self):
```

Comment:
this is shadowing a different method with the same name in the superclass. can you rename this one?